framework/deproxy_manager.py

```diff
@@ -55,9 +60,14 @@ def run_start(self):
         self.exit_event.clear()
         self.proc = threading.Thread(target = run_deproxy_server,
                                     args=(self, self.exit_event,
-                                          self.polling_lock))
+                                          self.polling_lock, self.exception))
         self.proc.start()
 
+    def thread_exception(self):
+        if self.exception.empty():
+            return None
+        return self.exception.get()
```

Comment:
According to Python docs `Queue.get()` against empty queue raises `Empty` exception, due to muly-consumer nature of the queue it's not guarantied that subsequent `get()` call won't issue an exception. At the moment the function is not used and there is nothing that guaranties that multiple consumer access happen.